src/components/auth/AuthModal.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useCallback, useState } from 'react';
+
+import clsx from 'clsx';
+import { Grow, Slide } from '@material-ui/core';
+
+import { useModal } from '@app/contexts/ui/Modal';
+
+import SignInWrapper from './SignInWrapper';
+import SignUp from './SignUp';
+
+import useStyles from './_jss/AuthModal.jss';
+
+enum SlidePosition {
+  RIGHT = 'right',
+  LEFT = 'left',
+}
+
+interface AuthModalProps {
+  /**
+   * The modal that will be face up on initial disclosure
+   */
+  initialVariant: 'sign-in' | 'sign-up';
+}
+
+const AuthModal: React.FC<AuthModalProps> = ({ initialVariant }) => {
+  /**
+   * The right slide is the "main" modal that shows face up on initial disclosure.
+   * The secondary modal slides in from the left on a toggle.
+   */
+  const [activeSlide, setActiveSlide] = useState<SlidePosition>(SlidePosition.RIGHT);
```

Comment:
I tried it both ways, this just felt better. It sort of establishes the concept of a "primary" and "secondary" screens and the secondary screen always slides in from the left.

Reply:
I'm not married to it, if you've got a decent case to make for static positioning I'd be willing to change it.

Reply:
Static positioning makes the most sense to me because the associated buttons are statically positioned. A user will feel more in control if their dialogue modals are not flipping sides between sessions.

Reply:
It's a small detail but the result is an experience that the user can anticipate. What makes one modal "primary" and the other "secondary"? Just whichever the user happened to click first?